lib/format.js

```diff
@@ -201,11 +217,52 @@ class Format {
   }
 
   buffer () {
+    if (this.parent && this.parent.isBuffer()) {
+      return this.style.bufferKey(this.key) +
+        this.style.bufferKeySep() +
+        this.style.bufferLine(this.object, this.bufferChunkSize)
+    }
+
     if (this.object.length === 0)
       return this.style.bufferEmpty()
-    return this.style.bufferStart() +
-      this.style.bufferBody(this.object) +
-      this.style.bufferEnd()
+
+    if (this.bufferIsShort()) {
+      return this.style.bufferStart() +
+        this.style.bufferBody(this.object) +
+        this.style.bufferEnd(this.object)
+    }
+
+    const b = this.bufferBody()
+    return this.bufferHead() + b + this.bufferTail()
+  }
+  isBuffer () {
+    return Buffer.isBuffer(this.object)
+  }
+  bufferIsShort () {
+    return this.object.length < this.bufferChunkSize + 5
```

Comment:
Should this be `this.object.length + 5 < this.bufferChunkSize`?  Can you add tests for expected output of the longest short buffer and the shortest long buffer?

Reply:
The 5 is arbitrary here, but it's to avoid a case where you have two lines, but the second line is only 1 byte.  Goes just a little bit longer than the chunk while still doing one line, so that if it does wrap, it doesn't feel pointless.